internal/ethapi/api.go

```diff
@@ -51,21 +51,32 @@ const defaultGas = uint64(90000)
 // It offers only methods that operate on public data that is freely available to anyone.
 type PublicEthereumAPI struct {
 	b Backend
+	solcPath *string
+	solc     **compiler.Solidity
 }
 
 // NewPublicEthereumAPI creates a new Etheruem protocol API.
-func NewPublicEthereumAPI(b Backend) *PublicEthereumAPI {
-	return &PublicEthereumAPI{b}
+func NewPublicEthereumAPI(b Backend, solcPath *string, solc **compiler.Solidity) *PublicEthereumAPI {
+	return &PublicEthereumAPI{b, solcPath, solc}
```

Comment:
Why are you passing in the instance as a double pointer? 
I think the only thing that needs to be passed here is the path as a plain string.
If the string is "", compiler.New will use the default path.